uniffi_bindgen/src/bindings/python/templates/RustBufferHelper.py

```diff
@@ -28,59 +24,230 @@ def _pack_into(self, size, format, value):
+    # Objects cannot currently be serialized, but we can produce a helpful error.
+
+    read{{ type_name|class_name_py }}(self):
+        raise RuntimeError("RustBufferStream.read() not implemented yet for {{ type_name }}")
+
+    write{{ type_name|class_name_py }}(self):
+        raise RuntimeError("RustBufferStream.write() not implemented yet for {{ type_name }}")
+
+    {% when Type::Error with (error_name) -%}
+    # The Error type {{ error_name }}.
+    # Errors cannot currently be serialized, but we can produce a helpful error.
+
+    read{{ type_name|class_name_py }}(self):
+        raise RuntimeError("RustBufferStream.read() not implemented yet for {{ type_name }}")
+
+    write{{ type_name|class_name_py }}(self):
+        raise RuntimeError("RustBufferStream.write() not implemented yet for {{ type_name }}")
+
+    {% when Type::Enum with (enum_name) -%}
+    # The Enum type {{ enum_name }}.
+
+    def read{{ type_name|class_name_py }}(self):
+        return {{ enum_name|class_name_py }}(
+            self._unpack_from(4, ">I")
+        )
+
+    def write{{ type_name|class_name_py }}(self, v):
+        self._pack_into(4, ">I", v.value)
+
+    {% when Type::Record with (record_name) -%}
+    # The Record type {{ record_name }}.
+
+    {% let rec = ci.get_record_definition(record_name).unwrap() %}
+
+    def read{{ type_name|class_name_py }}(self):
+        return {{ rec.name()|class_name_py }}(
+            {%- for field in rec.fields() %}
+            self.read{{ field.type_().unique_name()|class_name_py }}(){% if loop.last %}{% else %},{% endif %}
+            {%- endfor %}
+        )
+
+    def write{{ type_name|class_name_py }}(self, v):
+        {%- for field in rec.fields() %}
+        self.write{{ field.type_().unique_name()|class_name_py }}(v.{{ field.name() }})
+        {%- endfor %}
+
+    {% when Type::Optional with (inner_type) -%}
+    # The Optional<T> type for {{ inner_type.unique_name() }}.
+
+    def read{{ type_name|class_name_py }}(self):
+        flag = self._unpack_from(1, ">B")
+        if flag == 0:
+            return None
+        elif flag == 1:
+            return self.read{{ inner_type.unique_name()|class_name_py }}()
+        else:
+            raise RuntimeError("Unexpected flag byte for {{ type_name }}")
+
+    def write{{ type_name|class_name_py }}(self, v):
+        if v is None:
+            self._pack_into(1, ">B", 0)
+        else:
+            self._pack_into(1, ">B", 1)
+            self.write{{ inner_type.unique_name()|class_name_py }}(v)
+
+    {% when Type::Sequence with (inner_type) -%}
+    # The Sequence<T> type for {{ inner_type.unique_name() }}.
+
+    def read{{ type_name|class_name_py }}(self):
+        count = self._unpack_from(4, ">I")
+        items = []
+        while count > 0:
+            items.append(self.read{{ inner_type.unique_name()|class_name_py }}())
+            count -= 1
+        return items
+
+    def write{{ type_name|class_name_py }}(self, items):
```

Comment:
As a concrete example, in `rondpoint` this produces a `writeSequenceEnumEnumeration` method specialized for writing out sequences of `Enumeration` objects. We also get a `writeRecDictionnaire` method specialized for writing out `Dictionnaire` objects, etc.